src/Network/Network/LoadBalancer/NewAzureLoadBalancerBackendPool.cs

```diff
@@ -0,0 +1,167 @@
+// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using CNM = Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using System.Linq;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+
+namespace Microsoft.Azure.Commands.Network
+{
+    [Cmdlet(VerbsCommon.New, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "LoadBalancerBackendAddressPool"), OutputType(typeof(PSBackendAddressPool))]
+    public partial class NewAzureLoadBalancerBackendPool : NetworkBaseCmdlet
+    {
+        private const string CreateByNameParameterSet = "CreateByNameParameterSet";
+        private const string CreateByParentObjectParameterSet = "CreateByParentObjectParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The resource group name of the load balancer.",
+            ParameterSetName = CreateByNameParameterSet)]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the load balancer.",
+            ParameterSetName = CreateByNameParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string LoadBalancerName { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The load balancer resource.",
+            ValueFromPipeline = true,
+            ParameterSetName = CreateByParentObjectParameterSet)]
+        public PSLoadBalancer LoadBalancer { get; set; }
+
+        [Parameter(
+            Mandatory = true,
+            HelpMessage = "The name of the backend pool.")]
```

Comment:
do we need to specify all parameters set or by default is all?

Reply:
By default its all